### PR TITLE
Make ARM builder base more minimal

### PR DESCRIPTION
**Pull request by abhay-krishna**
Removing most of the binaries we don't need on ARM builder-base.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-distro-build-tooling/pull/254#" title="Author self-approved">abhay-krishna</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-distro-build-tooling).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-distro-build-tooling/blob/main/OWNERS)~~ [abhay-krishna]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
@abhay-krishna: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
builder-base-tooling-presubmit | 63b260eb2b5c14babe2e94274e3165a470aa5fee | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-distro-build-tooling/254/builder-base-tooling-presubmit/1456810330138087424) | `/test builder-base-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-distro-build-tooling&pr=254). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhay-krishna).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@abhay-krishna: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
builder-base-tooling-presubmit | aea4a4f8482675d772edc278c6986f8b621ecaac | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-distro-build-tooling/254/builder-base-tooling-presubmit/1456812733549776896) | `/test builder-base-tooling-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-distro-build-tooling&pr=254). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhay-krishna).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: builder-base/install.sh
Comment (jaxesn) [inline builder-base/install.sh:128]: This is too early for some of this to run. I think you can leave the yum and license stuff and move the go/pip clean up back towards the end.

**Review by jaxesn — CHANGES_REQUESTED** [review]
I wouldn't swear it to be the case, but I think yq is gonna be needed on the arm builder.

I'd probably also prefer to also update bash on the arm builder just so that is consistent between the two.

**Comment by jaxesn** [comment]
I'm also a bit worried about the golang stuff not being setup the same. It's totally fine not to have all the versions but the way this is written now go will not be in /root/sdk or whatever like it will be on the amd64 one. This may not be an issue but I'm not totally sure. We can def try it and adjust if not.

**Comment by abhay-krishna** [comment]
> I wouldn't swear it to be the case, but I think yq is gonna be needed on the arm builder.
> 
> I'd probably also prefer to also update bash on the arm builder just so that is consistent between the two. 

Awesome, I'll include these changes.

**Comment by abhay-krishna** [comment]
> I'm also a bit worried about the golang stuff not being setup the same. It's totally fine not to have all the versions but the way this is written now go will not be in /root/sdk or whatever like it will be on the amd64 one. This may not be an issue but I'm not totally sure. We can def try it and adjust if not. 

I'm not completely sure, but I think that go get where arm64 is failing first downloads the tarball for that Go version, maybe we could try converting some of these go calls into wget's to make it work? I'll look into it.

File: builder-base/install.sh
Comment (abhay-krishna) [reply builder-base/install.sh]: Done

**Review by jaxesn — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-distro-build-tooling/pull/254#" title="Author self-approved">abhay-krishna</a>*, *<a href="https://github.com/aws/eks-distro-build-tooling/pull/254#pullrequestreview-800536692" title="LGTM">jaxesn</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-distro-build-tooling).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-distro-build-tooling/blob/main/OWNERS)~~ [abhay-krishna,jaxesn]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
